src/frontend/src/expr/function_call.rs

```diff
@@ -114,10 +115,25 @@ impl FunctionCall {
             // types, they will be handled in `cast_ok`.
             return Self::cast_nested(child, target, allows);
         }
+        if child.is_unknown() {
```

Comment:
Could we be even more general and evaluate any constant expr?

Reply:
Old issue on const folding: https://github.com/risingwavelabs/risingwave/issues/4235

Reply:
I'm considering if it's cleaner to add an ExprMutator for PlanNode and add an optimization phase that
1. does constant folding
2. adds session timezone

Or, we can simply eval_const at `bind_expr`

Reply:
But anyway, it won't completely help us as now our expr framework will actually give an error when trying to directly cast string to timestamptz.

I think maintaining this Varchar const casting as a separate case is still useful.

Reply:
Anw, I think having this independent rule is fine as well.

Reply:
Actually I have debated with myself whether to leverage const folding to achieve the same. But as you also pointed out, it should be a limited version happening before expr rewriting, and we also need a true one after expr rewriting.

So I just decided to use this simpler one to serve as pre-rewrite eval, and make sure we do not have too many false warnings in v0.1.16 (to be frozen in 3 days). It is possible this special handling may not be necessary after our const folding is more powerful later.